test/ibm/runtime/test_runtime.py

```diff
@@ -1,907 +0,0 @@
-# This code is part of Qiskit.
-#
-# (C) Copyright IBM 2021.
```

Comment:
Thanks, this refactoring looks great. 

I personally also find another file name (`test/ibm/runtime/test_ibm_provider`) confusing. Why is it called provider while it includes tests for the `IBMRuntimeService` class? Maybe we can get rid of some legacy here as well?!

Reply:
oh definitely, I just haven't got to there yet. Although it'll probably be another PR as this one is already larger than I'd like.